fe/fe-core/src/main/java/com/starrocks/scheduler/mv/MVEpoch.java

```diff
@@ -0,0 +1,146 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.scheduler.mv;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.thrift.TMVEpoch;
+import lombok.Data;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+/**
+ * The incremental maintenance of MV consists of epochs, whose lifetime is defined as:
+ * 1. Triggered by transaction publish
+ * 2. Acquire the last-committed binlog LSN and latest binlog LSN
+ * 3. Start a transaction for incremental maintaining the MV
+ * 4. Schedule task executor to consume binlog since last-committed, and apply these changes to MV
```

Comment:
How to define how many offsets the binlog consumes?

Reply:
It's defined in `BinlogConsumeStateVO`. 
- Current consumption state is record in **VERSION** granularity, so a `Map<TableId, Version>` could record this information. This introduce some limitation of the 
- Fine-granularity consumption state could be binlog LSN of each tablet, but this could introduce a large number of metadata